test/libdnf/rpm/test_package.cpp

```diff
@@ -0,0 +1,346 @@
+    CPPUNIT_ASSERT_EQUAL(std::string("Fedora Project"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_packager());
+}
+
+
+void RpmPackageTest::test_get_vendor() {
+    CPPUNIT_ASSERT_EQUAL(std::string("Fedora Project"), get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_vendor());
+}
+
+
+void RpmPackageTest::test_get_url() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("http://cqrlib.sourceforge.net/"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_url()
+    );
+}
+
+
+void RpmPackageTest::test_get_summary() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("ANSI C API for quaternion arithmetic and rotation"),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_summary()
+    );
+}
+
+
+void RpmPackageTest::test_get_description() {
+    CPPUNIT_ASSERT_EQUAL(
+        std::string("CQRlib is an ANSI C implementation of a utility library for quaternion\n"
+            "arithmetic and quaternion rotation math."
+        ),
+        get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_description()
+    );
+}
+
+
+void RpmPackageTest::test_get_files() {
+    const auto files = get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_files();
+    // TODO no files in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, files);
+}
+
+
+void RpmPackageTest::test_get_provides() {
+    const auto provides = get_pkg("CQRlib-1.1.1-4.fc29.x86_64").get_provides();
+    const std::vector<std::string> expected = {
+        "libCQRlib.so.2()(64bit)",
+        "CQRlib = 1.1.1-4.fc29",
+        "CQRlib(x86-64) = 1.1.1-4.fc29",
+    };
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(provides));
+}
+
+
+void RpmPackageTest::test_get_requires() {
+    const auto reqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_requires();
+    const std::vector<std::string> expected = {"setup"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(reqs));
+}
+
+
+void RpmPackageTest::test_get_requires_pre() {
+    const auto prereqs = get_pkg("filesystem-3.9-2.fc29.x86_64").get_requires_pre();
+    // TODO no prereqs in test packages
+    const std::vector<std::string> expected = {};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(prereqs));
+}
+
+
+void RpmPackageTest::test_get_conflicts() {
+    const auto conflicts = get_pkg("nodejs-1:5.12.1-1.fc29.x86_64").get_conflicts();
+    const std::vector<std::string> expected = {"node <= 0.3.2-12"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(conflicts));
+}
+
+
+void RpmPackageTest::test_get_obsoletes() {
+    const auto obsoletes = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_obsoletes();
+    const std::vector<std::string> expected = {"npm < 3.5.4-6"};
+
+    CPPUNIT_ASSERT_EQUAL(expected, to_vector(obsoletes));
+}
+
+
+void RpmPackageTest::test_get_prereq_ignoreinst() {
+    const auto ipr = get_pkg("npm-1:5.12.1-1.fc29.x86_64").get_prereq_ignoreinst();
+    // TODO no prereq_ignoreinst (called Ipr: in libsolv testcases) in test packages
+    const std::vector<std::string> expected = {};
```

Comment:
What about to modify one RPM to have something to test?

Reply:
Do you know how to put this in a spec file? I'm unable to find this in documentation anywhere.

Reply:
Probably this explanation could help:
```
/**
 * dnf_package_get_prereq_ignoreinst
 * @pkg: a #DnfPackage instance.
 *
 * Returns safe to remove Requires(pre) and Requires(post) dependencies
 * of the package. Safe to remove means that the dependency is not needed by
 * Requires(preun) nor by Requires(postun).
 * The package must be installed otherwise empty list is returned.
 *
 * Returns: A #DnfReldepList
 *
 * Since: 0.45.1
 */
```

Reply:
Yeah, it does. The package needs to be installed though.